frontend/src/components/CreateEditSongbook.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Box,
+  Button,
+  Flex,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react";
+import { AxiosResponse } from "axios";
+import { useEffect, useState } from "react";
+import { UseAsyncReturn } from "react-async-hook";
+import { useNavigate } from "react-router-dom";
+import { Songbook } from "../models";
+import { createNewSongbook, editSongbook } from "../services/songs";
+
+interface CreateEditSongbookProps {
+  isOpen: boolean;
+  is_noodle_mode: boolean | undefined;
+  onClose: () => void;
+  asyncSongbook: UseAsyncReturn<AxiosResponse<Songbook, any>, never[]> | null;
+}
+
+export default function CreateEditSongbook({
+  isOpen,
+  onClose,
+  asyncSongbook,
+  is_noodle_mode,
+}: CreateEditSongbookProps) {
+  const [maxSongs, setMaxSongs] = useState<string>("");
+  const [title, setTitle] = useState<string>("");
+  const [actionVerb, setActionVerb] = useState<string>("DANCE");
+  const [isNoodleMode, setIsNoodleMode] = useState<boolean>(
+    is_noodle_mode || false
+  );
+  const [theme, setTheme] = useState<string>("");
+
+  useEffect(() => {
+    if (isOpen && asyncSongbook && asyncSongbook.result) {
+      if (asyncSongbook?.result?.data.max_active_songs !== null)
+        setMaxSongs(asyncSongbook?.result?.data.max_active_songs.toString());
+      setTitle(asyncSongbook?.result?.data.title);
+      setActionVerb(asyncSongbook?.result?.data.action_verb);
+      setIsNoodleMode(asyncSongbook?.result?.data.is_noodle_mode);
+      setTheme(asyncSongbook?.result?.data.theme);
+    }
+  }, [
+    isOpen,
+    asyncSongbook?.result?.data.max_active_songs,
+    asyncSongbook?.result?.data.title,
+    asyncSongbook?.result?.data.theme,
+    asyncSongbook?.result?.data.is_noodle_mode,
+    asyncSongbook?.result?.data.action_verb,
+  ]);
+
+  const navigate = useNavigate();
+  const parsedSongCap = parseInt(maxSongs);
+
+  const handleButtonClick = async (e) => {
+    e.preventDefault();
+    if (asyncSongbook === null) {
+      const result = await createNewSongbook(
+        maxSongs,
+        title,
+        actionVerb,
+        isNoodleMode,
+        theme
+      );
+      if (result !== false) {
+        navigate(`/live/${result.data.session_key}`);
+      } else {
+        console.log("Couldn't create new songbook.");
+      }
+    }
+    if (asyncSongbook && asyncSongbook.result) {
+      const result = await editSongbook(
+        asyncSongbook.result.data.session_key,
+        maxSongs,
+        title,
+        actionVerb,
+        isNoodleMode,
```

Comment:
Per IRL convo, since isNoodleMode can't be edited we can just remove it from the editSongbook params

Reply:
Removed!